src/NATS.Client.Core/Internal/HeaderWriter.cs

```diff
@@ -37,10 +67,7 @@ internal long Write(IBufferWriter<byte> bufferWriter, NatsHeaders headers)
                     var keySpan = bufferWriter.GetSpan(keyLength);
                     _encoding.GetBytes(kv.Key, keySpan);
                     if (!ValidateKey(keySpan.Slice(0, keyLength)))
-                    {
-                        throw new NatsException(
-                            $"Invalid header key '{kv.Key}': contains colon, space, or other non-printable ASCII characters");
-                    }
+                        throw new NatsException($"Invalid header key '{kv.Key}': contains colon, space, or other non-printable ASCII characters");
```

Comment:
1. @mtmk correct me if I'm off here, do we prefer including braces?
2.  If possible, this would be a good opportunity to wrap this `throw` into a helper method, e.x.

```
if (!ValidateKey(keySpan.Slice(0, keyLength)))
{
    NatsException.Throw($"Invalid header key '{kv.Key}': contains colon, space, or other non-printable ASCII characters")
}

public class NatsException : Exception
{
   //etc
   [MethodImpl(MethodImplOptions.NoInlining)]
   public static void Throw(string message)
   {
      throw new NatsException(message);
   }
   //etc
}
```

(There may be a throwhelper already floating around, if so prefer that)

Main advantage of such is we can get some better inlining/opts and avoid certain callstack checks on happy path....

Reply:
ouch just merged it. I shall open a new PR